src/manifest/period.ts

```diff
@@ -192,4 +186,102 @@ export default class Period {
       return ada.isSupported;
     });
   }
+
+  /**
+   * Get the last position where there shall be content
+   * on the period, whatever is the chosen adaptation
+   * @returns {null | undefined | number}
+   */
+  getContentEnd(): null | undefined | number {
+    let maximumPositionFromAdaptations: null | number = null;
+    const { video, audio } = this.adaptations;
+    if (video !== undefined) {
+      for (let i = 0; i < video.length; i++) {
+        const videoAdaptation = video[i];
+        const lastPosition = videoAdaptation.getContentEnd();
+        if (lastPosition === undefined) {
+          return undefined;
+        }
+        if (lastPosition === null) {
+          break;
+        }
+        if (isNullOrUndefined(maximumPositionFromAdaptations) ||
+            lastPosition < maximumPositionFromAdaptations) {
+          maximumPositionFromAdaptations = lastPosition;
+        }
+      }
+    }
+    if (audio !== undefined) {
+      for (let i = 0; i < audio.length; i++) {
+        const audioAdaptation = audio[i];
+        const lastPosition = audioAdaptation.getContentEnd();
+        if (lastPosition === undefined) {
+          return undefined;
+        }
+        if (lastPosition === null) {
+          break;
+        }
+        if (isNullOrUndefined(maximumPositionFromAdaptations) ||
+            lastPosition < maximumPositionFromAdaptations) {
+          maximumPositionFromAdaptations = lastPosition;
+        }
+      }
+    }
+    if (maximumPositionFromAdaptations === null) {
+      return null;
+    }
+    if (this.end === undefined) {
+      return maximumPositionFromAdaptations;
+    }
+    return Math.min(maximumPositionFromAdaptations, this.end);
+  }
```

Comment:
Kind of ugly why not relying to write code two times once for video and the second time for audio